### DISCUSSION TITLE
"Performers" and "Studio" must do it manualy

### DISCUSSION THREAD

**Discussion by kucingkembar** [stashapp/stash#4721] [Q&A] [answered]
hi, sorry for my bad English,
i using 
v0.25.1
Build hash:
bf7cb78d
Build time:
2024-03-13 03:30:08
version

1. I can get the "Studio Code" "Details", "Cover Image", "Tags"
But the other value like "Performers", "Movie", and "Title", you must be done manually:
select a scene, select edit, scrape with, confirm manually(Apply), Save,
can this procedure be done automatically?
Note: please do this to the pornstar/performers bio too

2. tool > Duplicate Scenes,
"Missing phashes for x scenes. Please run the phash generation task."
no matter I run "Phas generation", there are some x scenes that missing phashes,
can you add "scan for missing only"?

3. the tool > Scene Filename Parser
seems not to rename the filename,
what does this thing do?

~~4. I know some Python, and I like to rename based on {studio}_{title}-{performer}.{ext}
can I get this data using Python requests.get(http://localhost:9999/scenes/1)?~~

5. how to clean deleted files from the library?

**Comment by HereIam80** [comment, marked as answer]
Hey there.

1. This is not really true. What you get depends on the capability of the scraper you are using. You definitely can get performers and titles. Maybe those things can't be scraped from the specific site you are scraping from. Regardless, I would rather recommend using the tagger view and also using stashDB as a source if you have mostly main stream content. See this guide: https://docs.stashapp.cc/beginner-guides/guide-to-scraping/

2. If you are running the "Generate task" with the "Phash" option enabled and still have missing phashes by the end of it, that probably means that you have corrupted files, where ffmpeg is not able to read them completely (even though they may play fine). Also if you run a scan or generate task only the missing items are worked upon.

3. The filename parser is for setting metadata from your file name and not the other way around. Stash by itself without any plugins, doesn't move or rename any of your files.

5. Run the clean task.

**Comment by kucingkembar** [comment]
> Hey there.
> 
>     1. This is not really true. What you get depends on the capability of the scraper you are using. You definitely can get performers and titles. Maybe those things can't be scraped from the specific site you are scraping from. Regardless, I would rather recommend using the tagger view and also using stashDB as a source if you have mostly main stream content. See this guide: https://docs.stashapp.cc/beginner-guides/guide-to-scraping/
i studied this, maybe i selects
```
Skip matches that have more than one result
Skip single name performers with no disambiguation
```
so the performers not filled,
i start scan + generate now, the result may appear hours later
>     2. If you are running the "Generate task" with the "Phash" option enabled and still have missing phashes by the end of it, that probably means that you have corrupted files, where ffmpeg is not able to read them completely (even though they may play fine). Also if you run a scan or generate task only the missing items are worked upon.
it probably my videos broken like you said,
>     3. The filename parser is for setting metadata from your file name and not the other way around. Stash by itself without any plugins, doesn't move or rename any of your files.
Clear
>     4. Run the clean task.
Clear

**Comment by HereIam80** [comment]
You seem to talk about some automated task like "auto-tag" or "identify". The tagger view is manual and can be found here:
![image](https://github.com/stashapp/stash/assets/31324543/f0941351-af14-4758-a158-b90ffacab146)

Scan shouldn't take long since it skips files that are already in your database. And generate also only works on stuff that isn't already done. You could have a look at the logs in "settings => logs" to look up any errors that occur during a tasks.

**Comment by kucingkembar** [comment]
yes, i use both in settings:
**Identify**
Automatically set scene metadata using stash-box and scraper sources.
**Auto Tag**
Auto-tag content based on file paths.

then, i do what your show in attachment, but you must do it manually each "scene"
1. the query seems work, but you must edit to proper string
2. the scrape by fragment, usually produce "No results found" result

additional info : i tried uncheck
```
Skip matches that have more than one result
Skip single name performers with no disambiguation
```
but the result is still same

**Comment by kucingkembar** [comment]
Update : the "Performers" and "Studio" will not auto-added if the "Performers" and "Studio" do not exist 
(You Must Create "Performers" and "Studio" first)
it not final, i still study them

**Reply by shumhh** [reply]
Did you finally solve it? I've also been troubled by this problem
